### ISSUE TITLE
[🐞] Math.round is not serializing into q:func

### ISSUE THREAD

**Issue by mhevery** [QwikDev/qwik#4769]
### Which component is affected?

Qwik Optimizer (rust)

### Describe the bug

```typescript
import { component$, useSignal } from '@builder.io/qwik';

export default component$(() => {
  const count = useSignal(123);
  return (
    <>
      Count: {Math.round(count.value + 1)}
      <button onClick$={() => (count.value += 0.5)}>+1</button>
    </>
  );
});
```

The above should serialize into:
```html
<script q:func="qwik/json">
  document.currentScript.qFuncs=[
    (p0,)=>Math.round(p0.value)
  ]
</script>
```

But it does not. The result is that the component is being downloaded and re-rendered using the JSX path rather than the signal-dom path.

### Reproduction

https://stackblitz.com/edit/qwik-starter-v96ske?file=src%2Froutes%2Findex.tsx

### System Info

```shell
n/a
```

**Comment by gioboa** [comment]
@Varixo @wmertens is this case addressed with Qwik v2?

**Comment by Varixo** [comment]
> @Varixo @wmertens is this case addressed with Qwik v2?

I don't think so :/

**Comment by JerryWu1234** [comment]
Maybe I can try this one

**Comment by JerryWu1234** [comment]
I dug into this problem and discovered this bug in Rust. I struggle with Rust, so it takes me a long time.

**Comment by gioboa** [comment]
I tried with v2 and this is still unresolved.

**Comment by JerryWu1234** [comment]
@wmertens @Varixo 

need to discuss this further

**Comment by wmertens** [comment]
So the reason is that the optimizer doesn't know that Math has no side effects and is safe to move. 

I propose that the optimizer can assume that all globals are safe to move and it gets a new option to list unsafe globals.

**Comment by wmertens** [comment]
[Playground link](https://qwikdev-build-v2.qwik-8nx.pages.dev/playground/#f=Q0o0RmZYE40OovalsFUCCQpICW6LMFXD0MgY6BhcYeIMUm%2BlUO2bWJKhBy7kNcBG6AEzILBo1VYw1KyFhwC4aYKokashlqNqsFUw0DPVrLXTNoS3ZdCCRdN6NCkPvaQMAA), look at the client bundles

basically, moving globals into wrappedFn should be done. I think it's always safe.

**Comment by JerryWu1234** [comment]
@wmertens  Can I close this ?

**Comment by wmertens** [comment]
right, closing as not planned because of https://github.com/QwikDev/qwik/pull/8420#issuecomment-4096174025
